flink-cdc-connect/flink-cdc-source-connectors/flink-connector-postgres-cdc/src/test/java/com/ververica/cdc/connectors/postgres/source/PostgresSourceITCase.java

```diff
@@ -178,29 +192,39 @@ public void testJobManagerFailoverSingleParallelism() throws Exception {
     }
 
     @Test
-    public void testConsumingTableWithoutPrimaryKey() {
-        try {
+    public void testConsumingTableWithoutPrimaryKey() throws Exception {
+        if (scanStartupMode == DEFAULT_SCAN_STARTUP_MODE) {
+            try {
+                testPostgresParallelSource(
+                        1,
+                        scanStartupMode,
+                        FailoverType.NONE,
+                        FailoverPhase.NEVER,
+                        new String[] {"customers_no_pk"},
+                        RestartStrategies.noRestart());
+            } catch (Exception e) {
+                assertTrue(
+                        ExceptionUtils.findThrowableWithMessage(
+                                        e,
+                                        String.format(
+                                                "Incremental snapshot for tables requires primary key, but table %s doesn't have primary key",
+                                                SCHEMA_NAME + ".customers_no_pk"))
+                                .isPresent());
+            }
+        } else {
             testPostgresParallelSource(
```

Comment:
Other startup mode will not throw this exception?

Reply:
@ruanhang1993 , this exception is thrown when use primary keys to split table chunks. Other startup modes of Postgres cdc connector no need to generate snapshot split, thus no exception is thrown.
By the way, Postgres cdc connector now only has two startup modes: "initial" and "latest-offset". Thus, I just add test for  latest-offset  strategy.